service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerMidEventTest.java

```diff
@@ -34,31 +54,32 @@
     private static final byte[] PDF = {1, 2, 3, 4, 5};
 
     @MockBean
-    private UploadDocumentService uploadDocumentService;
+    private UploadDocumentService uploadService;
 
     @MockBean
-    private DocmosisDocumentGeneratorService documentGeneratorService;
+    private DocmosisDocumentGeneratorService generatorService;
+
+    @Captor
+    private ArgumentCaptor<DocmosisCaseManagementOrder> captor;
 
     ActionCaseManagementOrderControllerMidEventTest() {
         super("action-cmo");
     }
 
     @Test
-    void shouldAddDocumentReferenceToOrderAction() {
+    void shouldPostExpectedTemplateDataAndAddDocumentReferenceToOrderActionWhenGeneratingCaseManagementOrder() {
         Document document = document();
         DocmosisDocument docmosisDocument = new DocmosisDocument("case-management-order.pdf", PDF);
 
-        given(documentGeneratorService.generateDocmosisDocument(any(DocmosisData.class), any()))
-            .willReturn(docmosisDocument);
-        given(uploadDocumentService.uploadPDF(any(), any())).willReturn(document);
+        given(generatorService.generateDocmosisDocument(captor.capture(), any())).willReturn(docmosisDocument);
+        given(uploadService.uploadPDF(any(), any())).willReturn(document);
 
-        CaseDetails caseDetails = populatedCaseDetails();
-        caseDetails.getData().put(CASE_MANAGEMENT_ORDER_LOCAL_AUTHORITY.getKey(),
-            CaseManagementOrder.builder().build());
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(getCaseDetails());
 
-        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails);
+        verify(uploadService).uploadPDF(PDF, "draft-case-management-order.pdf");
 
-        verify(uploadDocumentService).uploadPDF(PDF, "draft-case-management-order.pdf");
+        //need to pass in captor for draft image string
+        assertThat(captor.getValue()).isEqualToComparingFieldByField(expectedTemplateData(captor.getValue()));
```

Comment:
I feel like there has to be nicer way of doing this, maybe in this assert we can ignore the draft image string and then have another assert to just check that it isn't empty

Reply:
Also either in this one or in #1088 whichever gets merged first this can then be changed as we will know the exact value

Reply:
this will go in first, I am happy to address separately, however I believe this is fine for now

Reply:
If this is being merged in first then it doesn't matter as we can just sort it out in that other story